src/App.jsx

```diff
@@ -14,33 +16,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function sortByParameter(goods, { param = '' }) {
+  let goodsArr = goods;
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  switch (param) {
+    case 'alphabetically': {
+      goodsArr = goods.sort((good1, good2) => {
+        return good1.localeCompare(good2);
+      });
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+      break;
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    case 'length': {
+      goodsArr = goods.sort((good1, good2) => {
+        return good1.length - good2.length;
+      });
+
+      break;
+    }
+
+    default: {
+      break;
+    }
+  }
+
+  return goodsArr;
+}
+
+function getGoodsFormatted(goods, reversed) {
+  if (reversed) {
+    return goods.reverse();
+  }
+
+  return goods;
+}
+
+export const App = () => {
+  const [sortBy, setSortBy] = useState('');
+  const [isReversed, setReversed] = useState(false);
+
+  const handleClick = param => {
+    setSortBy(param);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const obj = {
+    param: sortBy,
+  };
+
+  const goods = sortByParameter([...goodsFromServer], obj);
```

Comment:
```suggestion
  const goods = getSortedGoods(goodsFromServer, sortBy, isReversed);
```

Reply:
It doesn't work, because, after sorting, goodsFromServer changes and reset couldn't return it to previous input state.